modules/integration/tests-integration/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/header/HTTPResponseCodeTestCase.java

```diff
@@ -18,58 +18,92 @@
+	    OMElement apiConfig = util.loadResource(relativePath);
+	    addApi(apiConfig);
     }
 
-    @Test(groups = {"wso2.esb"}, description = "Test different response codes",
-          dataProvider = "getResponseCodes")
-    public void testReturnResponseCode(int responseCode) throws Exception {
-        int port = 9005;
-        String contentType = "text/xml";
-        SimpleWebServer simpleWebServer = new SimpleWebServer(port, responseCode);
-        log.info("Running the test for context type -" + contentType);
-        try {
-            simpleWebServer.start();
-            SimpleHttpClient httpClient = new SimpleHttpClient();
-            Map<String, String> headers = new HashMap<String, String>();
-            headers.put("Content-Type", contentType);
-            HttpResponse response = httpClient.doGet(context.getContextUrls().getServiceUrl() + "/passThoughProxy", headers);
-            log.info(response.getEntity().getContentType());
-            log.info(response.getStatusLine().getStatusCode());
-            simpleWebServer.terminate();
+	@Test(groups = { "wso2.esb" }, description = "Test different response codes", dataProvider = "getResponseCodes")
+	public void testReturnResponseCode(int responseCode) throws Exception {
+		int port = 8089;
+		server = HttpServer.create(new InetSocketAddress(port), 0);
+		server.createContext("/gettest", new MyHandler());
+		server.setExecutor(null); // creates a default executor
+		server.start();
+		switch (responseCode) {
+			case 200:
+				String contentType = "text/xml";
+				String url = "http://localhost:8280/serviceTest/test";
+				sendRequest(url, 200, contentType);
+			case 404:
+				contentType = "text/html";
+				url = "http://localhost:8280/serviceTest/notfound";
+				sendRequest(url, 404, contentType);
+		}
 
-            assertEquals(response.getEntity().getContentType().getValue(), contentType, "Expected content type doesn't match");
-            assertEquals(response.getStatusLine().getStatusCode(), responseCode, "response code doesn't match");
-        } finally {
-            simpleWebServer.terminate();
-            waitForPortCloser(port);
-        }
-    }
+		server.stop(0);
+
+	}
+
+	private class MyHandler implements HttpHandler {
+		public void handle(HttpExchange t) throws IOException {
+			Headers h = t.getResponseHeaders();
+			h.add("Content-Type", "text/xml");
+			String response = "This is Response status code test case";
+			t.sendResponseHeaders(200, response.length());
+			OutputStream os = t.getResponseBody();
+			os.write(response.getBytes());
+			os.close();
+		}
+	}
+
+	private void sendRequest(String url, int responseCode, String contentType) {
+		DefaultHttpClient httpclient = new DefaultHttpClient();
+		HttpGet httpGet = new HttpGet(url);
+		HttpResponse response = null;
+		try {
+			response = httpclient.execute(httpGet);
+		} catch (IOException e) {
+			log.error("Error Occured while sending http get request. " + e);
```

Comment:
Better not to concatenate exception object to the message. It can be passed as the second parameter: log.error("message", e)